datahub-classify/src/datahub_classify/helper_classes.py

```diff
@@ -1,39 +1,146 @@
+        default=None,
+        description="confidence score using platform. For columns, it is None",
+    )
+    table_schema: Optional[float] = Field(
+        default=None,
+        description="For tables, it is confidence score using table schema. For columns, it is table_similarity_score",
+    )
+    lineage: Optional[float] = Field(
+        default=None, description="confidence score using lineage"
+    )
+
+
+class SimilarityDebugInfo(BaseModel):
+    name: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="confidence score and weighted score contribution using name",
+    )
+    description: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="confidence score and weighted score contribution using description",
+    )
+    datatype: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="confidence score and weighted score contribution using datatype. For tables, it is None",
+    )
+    values: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="confidence score and weighted score contribution using values. For tables, it is None",
+    )
+    platform: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="confidence score and weighted score contribution using platform. For columns, it is None",
+    )
+    table_schema: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="For tables, it is confidence score and weighted score contribution using table schema. For columns, it is table_similarity_score",
+    )
+    lineage: Optional[FactorDebugInfo] = Field(
+        default=None,
+        description="confidence score and weighted score contribution using lineage",
+    )
+
 
 @dataclass
 class InfotypeProposal:
     infotype: str
     confidence_level: float
-    debug_info: Dict[str, Any]
+    debug_info: DebugInfo
+
+
+@dataclass
+class TextEmbeddings:
+    emb_type: str
+    embedding: numpy.ndarray
 
 
 @dataclass
-class Metadata:
-    meta_info: Dict[str, Any]
-    name: str = field(init=False)
-    description: str = field(init=False)
-    datatype: str = field(init=False)
-    dataset_name: str = field(init=False)
+class ColumnMetadata:
+    meta_info: InitVar[Optional[Dict[str, Any]]] = None
+    name: Optional[str] = None
+    description: Optional[str] = None
+    datatype: Optional[str] = None
+    dataset_name: Optional[str] = None
+    column_id: Optional[str] = None
+    name_embedding: List[TextEmbeddings] = field(default_factory=list)
+    desc_embedding: List[TextEmbeddings] = field(default_factory=list)
 
-    def __post_init__(self):
-        self.name = self.meta_info.get("Name", None)
-        self.description = self.meta_info.get("Description", None)
-        self.datatype = self.meta_info.get("Datatype", None)
-        self.dataset_name = self.meta_info.get("Dataset_Name", None)
+    def __post_init__(self, meta_info):
```

Comment:
We can avoid object creation using meta_info dictionary. This change was thought about earlier but we kept this way of object creation for the backward compatibility. Since we are not worried about that now, we will remove the usage of meta_info and will have a cleaner implementation.

Reply:
Removed meta info from the input parameters and restructured the code.